From 452dd3af319d06c09a79b9eedec3bd414243a41e Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Thu, 23 Jan 2020 15:56:43 +0200
Subject: [PATCH] Add additional include path for DRM

Tegra has drm.h header in include/drm instead of include/libdrm

Both paths are defined in libdrm.pc, but those are not used since this
cmake file searches only for xf86drm.h.

Change-Id: If1e979c7d5aec520b18eed5b3fcbb5ac2e15cc62
---
 cmake/3rdparty/kwin/FindLibdrm.cmake | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/cmake/3rdparty/kwin/FindLibdrm.cmake b/cmake/3rdparty/kwin/FindLibdrm.cmake
index 9936e07eee..10822e693a 100644
--- a/cmake/3rdparty/kwin/FindLibdrm.cmake
+++ b/cmake/3rdparty/kwin/FindLibdrm.cmake
@@ -105,13 +105,19 @@ if(NOT WIN32)
             INTERFACE_INCLUDE_DIRECTORIES "${Libdrm_INCLUDE_DIR}"
             INTERFACE_INCLUDE_DIRECTORIES "${Libdrm_INCLUDE_DIR}/libdrm"
         )
+        if(EXISTS "${Libdrm_INCLUDE_DIR}/drm")
+            set_property(TARGET Libdrm::Libdrm APPEND PROPERTY
+                INTERFACE_INCLUDE_DIRECTORIES "${Libdrm_INCLUDE_DIR}/drm"
+            )
+        endif()
+
     endif()
 
     mark_as_advanced(Libdrm_LIBRARY Libdrm_INCLUDE_DIR)
 
     # compatibility variables
     set(Libdrm_LIBRARIES ${Libdrm_LIBRARY})
-    set(Libdrm_INCLUDE_DIRS ${Libdrm_INCLUDE_DIR} "${Libdrm_INCLUDE_DIR}/libdrm")
+    set(Libdrm_INCLUDE_DIRS ${Libdrm_INCLUDE_DIR} "${Libdrm_INCLUDE_DIR}/libdrm" "${Libdrm_INCLUDE_DIR}/drm")
     set(Libdrm_VERSION_STRING ${Libdrm_VERSION})
 
 else()
